src/number_theory/quadratic_reciprocity.lean

```diff
@@ -418,19 +418,23 @@ begin
   simp [*, ne_neg_self p one_ne_zero, (ne_neg_self p one_ne_zero).symm] at *
 end
 
-lemma legendre_sym_eq_one_iff {a : ℕ} (ha0 : (a : zmod p) ≠ 0) :
+lemma legendre_sym_eq_one_iff {a : ℤ} (ha0 : (a : zmod p) ≠ 0) :
   legendre_sym p a = 1 ↔ (∃ b : zmod p, b ^ 2 = a) :=
 begin
-  rw [euler_criterion p ha0, legendre_sym, int.cast_coe_nat, if_neg ha0],
+  rw [euler_criterion p ha0, legendre_sym, if_neg ha0],
   split_ifs,
   { simp only [h, eq_self_iff_true] },
   { simp only [h, iff_false], tauto }
 end
 
 lemma eisenstein_lemma [fact (p % 2 = 1)] {a : ℕ} (ha1 : a % 2 = 1) (ha0 : (a : zmod p) ≠ 0) :
```

Comment:
Should this become an integer too?

Reply:
I have left this, since the proof does not seem to be easily adaptable. The result is used for proving quadratic reciprocity, wheere the arguments are (natural) prime numbers, and there is probably not much use outside of this.